lightning/src/routing/network_graph.rs

```diff
@@ -428,6 +428,70 @@ impl Readable for NodeInfo {
 	}
 }
 
+/// Allows for updating user's network metadata.
+pub trait UpdateMetadata {
+	/// Adds a failed route to track. If we see the route again, we'll reject it or we can take
+	/// some action (like set its fee to make route-finding not find that route again)
+	fn add_failed_route(&mut self, route: Vec<Vec<RouteHop>>);
+	/// If a route starts working again, then we can take it off the failed route list.
+	fn remove_failed_route(&mut self, route: Vec<Vec<RouteHop>>);
+	/// Returns true if a route is on the failed routes list, false otherwise.
+	fn check_route(&self, route: Vec<Vec<RouteHop>>) -> bool;
+	/// Sets the channel fee penalty being tracked in the Metadata object.
+	fn set_channel_fee_penalty(&mut self, chan_id: u64, fee_penalty: u64);
```

Comment:
Same here - rust-lightning will never call this so it doesnt need to be in the trait.